frontend/src/contexts/BookContext.js

```diff
@@ -0,0 +1,52 @@
+
+import React, { createContext, useState, useEffect } from 'react';
+
+const BookContext = createContext();
+
+export const BookProvider = ({ children }) => {
+    const [purchasedBooks, setPurchasedBooks] = useState(() => {
+        return JSON.parse(localStorage.getItem('purchasedBooks')) || [];
+    });
+
+    useEffect(() => {
+        localStorage.setItem('purchasedBooks', JSON.stringify(purchasedBooks));
```

Comment:
always add try catch for api use
like
try {
      localStorage.setItem('purchasedBooks', JSON.stringify(purchasedBooks));
    } catch (error) {
      console.error('Error saving to localStorage:', error);
    }